pageserver/src/tenant/timeline.rs

```diff
@@ -455,6 +455,12 @@ pub(crate) struct GcInfo {
 
     /// The cutoff coordinates, which are combined by selecting the minimum.
     pub(crate) cutoffs: GcCutoffs,
+
+    // TODO(yuchen): If we decide to incorporate `retain_lsns` into the same structure,
+    // this would look something like `BTreeMap<Lsn, Option<LsnLease>>`, where
+    // `None` suggests an infinite lease (will be used by current retain_lsns).
+    /// Leases granted to particular LSNs.
+    pub(crate) leases: BTreeMap<Lsn, LsnLease>,
```

Comment:
Still investigating ways to merge leases with `retain_lsns` and whether it will be a huge refactor.

Reply:
For clarity, I will keep a separate leases map for now.